src/main/webapp/app/entities/participant/participant.component.ts

```diff
@@ -43,6 +43,34 @@ export class ParticipantComponent implements OnInit, OnDestroy {
         this.reverse = true;
     }
 
+    /**
+     * This function returns the highest number of Repositories that any user has for limiting values in the
+     * participant filtering system.
+     * */
+    getMaxRepositories(){
+        let  highestFound = 0;
+        for (let participant of this.participants){
+            if(participant.numberOfRepositories > highestFound){
+                highestFound = participant.numberOfRepositories;
+            }
+        }
+        return highestFound;
+    }
+
+    /**
+     * This function returns the highest number of contributions that any user has for limiting values in the
+     * participant filtering system.
+     * */
+    getMaxContributions(){
+        let  highestFound = 0;
+        for (let participant of this.participants){
+            if(participant.numberOfContributions > highestFound){
+                highestFound = participant.numberOfContributions;
+            }
+        }
+        return highestFound;
+    }
+
```

Comment:
Both methods seem to have similar behaviour of returning the maximum. Can this be refactored into a common max() method or use a library call. 
I think this works, it compiles but I haven't tested it.
`let repos = this.participants.map(p => p.numberOfRepositories);`
 `return repos.reduce((a,b) => Math.max(a,b));`

Reply:
Thanks, didn't know about Typescript reduce. I'll use it

Reply:
Yeah both methods are just getting max.

The method is called inside the html with the returned value use as the maximum upper bound for the field.

The only to squash both methods into one would be to pass a flag when called and use an or depending on which flag was passed. There would still be code duplication with the or branches.

I feel flag passing will get cumbersome if there are more numerical fields added in the future (related to ticket #41) and open the potential for bugs as it means working methods will have to be edited potentially breaking them.

Reply:
getMaxRepositories(){
        return getMax(this.partcipants.map(p => p.numberOfRepositories);
    }

    getMaxContributions(){
        return getMax(this.partcipants.map(p => p.numberOfContributions);
    }

    getMax(arr: Number[]){
        return arr.reduce((a,b) => Math.max(a,b));
    }

Try this?

Reply:
Just read a similar example online saying this is bad because it mixes abstraction levels but I think that's being too pedantic.